src/Operation/FindAndModify.php

```diff
@@ -243,6 +253,17 @@ public function execute(Server $server)
 
         $cursor = $server->executeWriteCommand($this->databaseName, new Command($this->createCommandDocument()), $this->createOptions());
 
+        if (isset($this->options['codec'])) {
+            $cursor->setTypeMap(['root' => 'bson']);
+            $result = current($cursor->toArray());
+            assert($result instanceof Document);
+
+            $decoded = $this->options['codec']->decodeIfSupported($result->get('value') ?? null);
```

Comment:
Note that a bit later in this method we do `$result->value ?? null` to gracefully handle a case where `value` does not exist. I'm not sure what the point of a [null-coalescing operator](https://www.php.net/manual/en/language.operators.comparison.php#language.operators.comparison.coalesce) is for `Document::get()`, since the method throws if the field doesn't exist.

[findAndModify: value](https://www.mongodb.com/docs/manual/reference/command/findAndModify/#std-label-findandmodify-value-field) suggests that the field will always be either a BSON object or null, so perhaps this was only needed for older server versions. If we're not able to confirm that, I might suggest adding some common code to gracefully extract `value` here.

I don't think we can do it for a common case earlier in this method because the `typeMap` option handling requires application to the cursor before we access the value.

Reply:
That was a brainfart on my part. I've removed the null-coalesce from this.